### PR TITLE
Remove policy exemption and tweak some model names

### PR DESCRIPTION
**Pull request by ArcturusZhang**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by ArthurMa1978 — APPROVED** [review]

**Review by Yao725 — APPROVED** [review]
LGTM, but please also disable the corresponding tests.

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupData.cs [hunk 1]
```diff
@@ -12,12 +12,12 @@
 namespace Azure.ResourceManager.ManagementGroups.Models
 {
     /// <summary> Management group creation parameters. </summary>
-    public partial class CreateManagementGroupOptions
+    public partial class CreateManagementGroupData
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupData.cs:15]: We should not use the suffix Data unless it inherits from one of our ResourceData objects, we can just undo this rename

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupChildInfo.Serialization.cs [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 namespace Azure.ResourceManager.ManagementGroups.Models
 {
-    public partial class ManagementGroupChildOptions
+    public partial class CreateManagementGroupChildInfo
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupChildInfo.Serialization.cs:10]: undo this rename

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupChildInfo.cs [hunk 3]
```diff
@@ -11,12 +11,12 @@
 namespace Azure.ResourceManager.ManagementGroups.Models
 {
     /// <summary> The child information of a management group used during creation. </summary>
-    public partial class ManagementGroupChildOptions
+    public partial class CreateManagementGroupChildInfo
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupChildInfo.cs:14]: undo this rename

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupData.Serialization.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@
 
 namespace Azure.ResourceManager.ManagementGroups.Models
 {
-    public partial class CreateManagementGroupOptions : IUtf8JsonSerializable
+    public partial class CreateManagementGroupData : IUtf8JsonSerializable
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupData.Serialization.cs:13]: undo this rename

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateParentGroupInfo.Serialization.cs [hunk 5]
```diff
@@ -10,7 +10,7 @@
 
 namespace Azure.ResourceManager.ManagementGroups.Models
 {
-    public partial class ManagementGroupParentCreateOptions : IUtf8JsonSerializable
+    public partial class CreateParentGroupInfo : IUtf8JsonSerializable
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateParentGroupInfo.Serialization.cs:13]: undo this rename

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateParentGroupInfo.cs [hunk 6]
```diff
@@ -8,10 +8,10 @@
 namespace Azure.ResourceManager.ManagementGroups.Models
 {
     /// <summary> (Optional) The ID of the parent management group used during creation. </summary>
-    public partial class ManagementGroupParentCreateOptions
+    public partial class CreateParentGroupInfo
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateParentGroupInfo.cs:11]: undo this rename

**Review by m-nash — CHANGES_REQUESTED** [review]

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupData.cs (hunk 1 above)
Comment (m-nash) [reply sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/Models/CreateManagementGroupData.cs:15]: [Entity][Method]Options -> reserved for optional parameter bag

File: sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/ManagementGroupCollection.cs [hunk 7]
```diff
@@ -64,22 +64,22 @@ internal static void ValidateResourceId(ResourceIdentifier id)
...
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentException"> <paramref name="groupId"/> is an empty string, and was expected to be non-empty. </exception>
-        /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="options"/> is null. </exception>
-        public virtual async Task<ArmOperation<ManagementGroupResource>> CreateOrUpdateAsync(WaitUntil waitUntil, string groupId, CreateManagementGroupOptions options, string cacheControl = null, CancellationToken cancellationToken = default)
+        /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="data"/> is null. </exception>
+        public virtual async Task<ArmOperation<ManagementGroupResource>> CreateOrUpdateAsync(WaitUntil waitUntil, string groupId, CreateManagementGroupData data, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (m-nash) [inline sdk/resourcemanager/Azure.ResourceManager/src/ManagementGroup/Generated/ManagementGroupCollection.cs:72]: CreateManagementGroupOptions
ManagementGroupCreateOrUpdateOptions

CreateManagementGroupInfo

Patchable[Entity]Data
Creatable[Entity]Data

[Entity]CreateOrUpdateInfo
[Entity][Method]Info

**Review by m-nash — APPROVED** [review]
Approve based on undoing the renames and moving that into a separate PR

**Pull request merged**